scene/gui/line_edit.cpp

```diff
@@ -2262,11 +2263,27 @@ bool LineEdit::is_drag_and_drop_selection_enabled() const {
 	return drag_and_drop_selection_enabled;
 }
 
+void LineEdit::_texture_changed() {
+	_fit_to_width();
+	update_minimum_size();
+	queue_redraw();
+}
+
 void LineEdit::set_right_icon(const Ref<Texture2D> &p_icon) {
 	if (right_icon == p_icon) {
 		return;
 	}
+
+	if (right_icon.is_valid()) {
+		right_icon->disconnect(CoreStringNames::get_singleton()->changed, callable_mp(this, &LineEdit::_texture_changed));
```

Comment:
Any reason not to use `(dis)connect_changed`? Otherwise it is:
```suggestion
		right_icon->disconnect(CoreStringName(changed), callable_mp(this, &LineEdit::_texture_changed));
```
For all

Reply:
simply was not aware of it